tcases-lib/src/main/java/org/cornutum/tcases/io/ObjectUtils.java

```diff
@@ -0,0 +1,172 @@
+//////////////////////////////////////////////////////////////////////////////
+// 
+//                    Copyright 2018, Cornutum Project
+//                             www.cornutum.org
+//
+//////////////////////////////////////////////////////////////////////////////
+
+package org.cornutum.tcases.io;
+
+import java.math.BigDecimal;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * Defines utility methods for reading and writing objects.
+ *
+ */
+public final class ObjectUtils
+  {
+  private ObjectUtils()
+    {
+    // Static methods only
+    }
+  
+  /**
+   * Returns the object represented by the given string.
+   */
+  public static Object toObject( String value)
+    {
+    return toObject( value, toBoolean, toNumber, toString);
+    }
+  
+  
+  /**
+   * Returns the first non-null object produced by the given converters.
+   * Returns null if no such object found.
+   */
+  @SafeVarargs
+  private static <T> Object toObject( T value, Function<T,Object>... converters)
+    {
+    return
+      Stream.of( converters)
+      .map( converter -> converter.apply( value))
+      .filter( object -> object != null)
+      .findFirst()
+      .orElse( null);
+    }
+
+  /**
+   * Returns the Boolean represented by the given string, or null if not a boolean value.
+   */
+  private static Function<String,Object> toBoolean =
+    new Function<String,Object>()
+      {
+      public Object apply( String value)
+        {
+        return
+          "true".equalsIgnoreCase( value)?
+          Boolean.TRUE :
+
+          "false".equalsIgnoreCase( value)?
+          Boolean.FALSE :
+
+          null;
+        }
+      };
+
+  /**
+   * Returns the given String value.
+   */
+  private static Function<String,Object> toString =
+    new Function<String, Object>()
+      {
+      public Object apply( String value)
+        {
+        // "null" is string representation of null
+        return
+          "null".equals( value)
+          ? null
+          : value;
+        }
+      };
+
+  /**
+   * Returns the Number represented by the given string, or null if not a numeric value.
+   */
+  private static Function<String,Object> toNumber =
+    new Function<String,Object>()
+      {
+      public Object apply( String value)
+        {
+        Object number;
+
+        try
+          {
+          number = toObject( new BigDecimal( value), toInt, toLong, toDecimal);
+          }
+        catch( Exception e)
```

Comment:
maybe if the exception handling happened inside the toObject Stream, the functions here would be much shorter?

Reply:
This is handling the NumberFormatException that could be thrown by the BigDecimal constructor.

Reply:
Another way to shorten the code is to wrap in a function `nullOnException( Function<String,Object> fun)`